### DISCUSSION TITLE
Specific logger per Session

### DISCUSSION THREAD

**Discussion by MaartenUreel** [sqlalchemy/sqlalchemy#9640] [Usage Questions] [answered]
I want to assign a specific logger to every Session that I created in SQLAlchemy.

The logger is one that writes to a specific stream within the context of a phonecall running on FreeSWITCH.

Is there a way I can specify a specific logger (LoggerAdapter) to the Session so that I can echo all the queries to there?

**Comment by zzzeek** [comment, marked as answer]
first off Sessions dont have any logging functions built in.   I think you are likely referring to the engine, which logs on the `sqlalchemy.engine` logging namespace.   This namespace is hardcoded and cannot be changed.  However, the log messages themselves can be augmented with additional sub-engines as detailed in the section [Setting per-connection  / sub-engine tokens](https://docs.sqlalchemy.org/en/20/core/engines.html#setting-per-connection-sub-engine-tokens).

if you needed these messages to be routed in some special way you could implement a custom [logging handler](https://docs.python.org/3/library/logging.handlers.html) that can multiplex messages based on tokens to different locations.
